### PR TITLE
Enable Intersection Observer object by default

### PR DESCRIPTION
**Pull request by nrajan002c**
Signed-off-by: nganes961 <nambirajan_ganesan@comcast.com>

### CODE REVIEW

**Review by emutavchi — COMMENTED** [review]
LGTM, need someone from Metro to review as well

File: WebKitBrowser/WebKitImplementation.cpp [hunk 1]
```diff
@@ -2275,6 +2275,9 @@ static GSourceFuncs _handlerIntervention =
             // Turn on/off local storage
             WKPreferencesSetLocalStorageEnabled(preferences, _localStorageEnabled);
 
+            // Turn on Intersection Observer object
```
Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2278]: Could you please only enable this for WebKit 2.28 or later (#ifdef)?

Indeed there seems to be no side effect to turn it on, so it does not have to be configurable, but it is only implemented in 2.28 and as the plugin also needs to work with older webkit versions (it seems to be implemented empty in 2.22 but might not be available at all in older versions) better to have it only enabled on 2.28 or later.

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: WebKitBrowser/WebKitImplementation.cpp (hunk 1 above)
Comment (nrajan002c) [reply WebKitBrowser/WebKitImplementation.cpp:2278]: Hi @MFransen69 
We are currently requiring this support in 2.22 for one of the premium app. 
We had similar issue with other observer  (ResizeObserver) support was not there in 2.22. We had brought to this branch. 
Could you please share the IntersectionObserver changes, we can add this to 2.22 to make it clear.

Comment (MFransen69) [reply WebKitBrowser/WebKitImplementation.cpp:2278]: Could you please reach out to Özgür Deveci for that (@modeveci)? Think then still you would need to conditionally add the call here for older WebKit versions (like 2017) but that you can also ask him. Thanks!

**Pull request closed without merging**